### PR TITLE
Update preprocessing module to consider both 1.0 and 0.25 degrees

### LINKED ISSUE #10: Utility to create GFS and ERA5 data in 1.0-degree resolution.
A utility is required for generating training data of ML4BC at 1.0-degree spatial, 3-hourly temporal, and 16-day lead time. All relevant variables for the 2m temperature bias correction need to be considered.

### PR DESCRIPTION
**Pull request by sadeghitabas**
## Description
This PR provides two Python scripts for generating training data for ML4BC in two 1.0- and 0.25-degree spatial resolutions, respectively.  
The extracted GFS variables include:

1. 2 metre temperature, 
2.  Temperature  at surface, 
3.  Temperature at 1000 mb, 
4.  Relative humidity at 1000 mb, 
5.  Pressure reduced to MSL, 
6.  10 metre U wind component, 
7.  10 metre V wind component,
8.  Soil temperature at top layer
9.  Liquid volumetric soil moisture (non-frozen) at top layer
10.  Land-Sea Mask

From ERA5, the utility extracts 2 metre temperature as the ground truth information. 

Newly added functionalities include:
- `gen_training_1.0d.py`: script to generate GFS and ERA5 forecasts required for ML4BC training at 1.0-degree spatial and 3-hourly temporal resolutions (lead time: 16 days).
- `gen_training_0.25d.py`: script to generate GFS and ERA5 forecasts required for ML4BC training at 0.25-degree spatial and hourly temporal resolutions (lead time: 50 hours).
- `jobcard_submission.py`: script to make and submit ERA5 and GFS preprocessing job cards in parallel.

Note: The above utilities are flexible in updating the variables by specifying the variable name, type of level, and level value to any arbitrary values that exist in the resources.

### Associated GraphCast Issue to close
- Closes https://github.com/NOAA-EMC/ML4BC/issues/10

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [ ] New input data.
  - [x] Updated input data.

### Needed libraries

  - `python3`
  - `xarray`
  - `boto3`
  - `netCDF4`
  - `pygrib`
  - `cdsapi`
  - `matplotlib`
  - `ipywidgets`

### CODE REVIEW

File: gen_training_0.25d.py [hunk 1]
```diff
@@ -240,8 +260,20 @@ def era5_process(self):
...
-    data_processor = DataProcessor(start_date, end_date, output_directory, download_directory, keep_downloaded_data)      
+    
+    gfs_variables_with_levels = {
+        '2 metre temperature': [{'typeOfLevel': 'heightAboveGround', 'level': 2}],
+        'Temperature': [{'typeOfLevel': 'isobaricInhPa', 'level': 1000},{'typeOfLevel': 'surface', 'level': 0}],
+        'Relative humidity': [{'typeOfLevel': 'isobaricInhPa', 'level': 1000}],
+        'Pressure reduced to MSL': [{'typeOfLevel': 'meanSea', 'level': 0}],
+        '10 metre U wind component': [{'typeOfLevel': 'heightAboveGround', 'level': 10}],
+        '10 metre V wind component': [{'typeOfLevel': 'heightAboveGround', 'level': 10}],
+        'Soil Temperature': [{'typeOfLevel': 'depthBelowLandLayer', 'level': 0}],
```
Comment (junwang-noaa) [inline gen_training_0.25d.py:271]: Can you double check this , usually we have this "0-0.1 m"
827:50829985:d=2021032206:TSOIL:0-0.1 m below ground:24 hour fcst:

Comment (sadeghitabas) [reply gen_training_0.25d.py:271]: this gives me "564:Soil temperature:K (instant):regular_ll:depthBelowLandLayer:levels 0.0-0.1 m:fcst time 0 hrs:from 202203030000" 
is it the wrong one?

Comment (junwang-noaa) [reply gen_training_0.25d.py:271]: OK, this is the correct one. Thanks for confirm.

**Review by junwang-noaa — APPROVED** [review]

File: gen_training_0.25d.py (hunk 1 above)
Comment (sadeghitabas) [reply gen_training_0.25d.py:271]: I'm loading them by the keys using the pygrib library, its a little bit different from wgrib2. Also, when I open a GFS grib2 file using pygrib we have 696 records, idk why yours showing #827. Thanks

Comment (junwang-noaa) [reply gen_training_0.25d.py:271]: I was looking at the field in a GFSv7 regression test grib file, which is different from the GSFv16 opn files.
